src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx

```diff
@@ -450,7 +450,7 @@ export class ComparisonFilter extends React.Component<ComparisonFilterProps, Com
     const {
       onFilterChange
     } = this.props;
-    let filter: GsComparisonFilter = _cloneDeep(this.state.filter);
+    let filter: GsComparisonFilter = this.state.filter;
     filter[2] = newValue;
```

Comment:
Settings the newValue (line 454) will manipulate the JS object in place. This might lead to problems when rerendering the component and comparing its previous value with the new one set via `setState`.

Maybe we can get rid of setting the value but calling setState with the correct value. Can you try to get it working with line [467](https://github.com/geostyler/geostyler/pull/1345/files#diff-98cfd035aec1fee5bb85b5e61c3949d86365efe3e6e09484cbe6a62363724f1fL467) changed to something like this:

```js
filter: [filter[0], filter[1], newValue]
```

Untested code. :smile:

Reply:
I did it like this:
```javascript

// let filter: GsComparisonFilter = this.state.filter;
// filter[1] = newAttrName;

const filter: GsComparisonFilter = [
    this.state.filter[0], 
    newAttrName, 
    this.state.filter[2]
  ];
```

@KaiVolland is that what you meant?

Unfortunately it does not fix the bug.

Reply:
Yes that is what i mean but i think it should be:

```js
// let filter: GsComparisonFilter = this.state.filter;
// filter[1] = newAttrName;

const filter: GsComparisonFilter = [
    this.state.filter[0], 
    this.state.filter[1]
    newValue
  ];
```

If this does not work we can still merge your base approach.

Reply:
A "raw" example of a ComparisonFilter is:

```javascript
      filter: [
        "==",
        "my_attribute",
        "my_value"
      ]
```

The "attribute" property has to be at the second position. See also the definition in the [geostyler-style repo](https://github.com/geostyler/geostyler-style/blob/master/index.d.ts#L62-L71).

Reply:
Also tested the approach suggested by @KaiVolland but unfortunately it still leads to the error with the cursor shift.

Reply:
> A "raw" example of a ComparisonFilter is:
> 
> ```js
>       filter: [
>         "==",
>         "my_attribute",
>         "my_value"
>       ]
> ```
> 
> The "attribute" property has to be at the second position. See also the definition in the [geostyler-style repo](https://github.com/geostyler/geostyler-style/blob/master/index.d.ts#L62-L71).

Strange, as the previous code sets the value at index 2. But if it works… :) I'll have a look at this later.